aws-opentelemetry-distro/src/amazon/opentelemetry/distro/patches/_botocore_patches.py

```diff
@@ -29,6 +30,27 @@ def _apply_botocore_instrumentation_patches() -> None:
     _apply_botocore_s3_patch()
     _apply_botocore_sqs_patch()
     _apply_botocore_bedrock_patch()
+    _apply_botocore_sns_patch()
+
+
+def _apply_botocore_sns_patch() -> None:
+    """Botocore instrumentation patch for SNS
+
+    This patch extends the existing upstream extension for SNS. Extensions allow for custom
+    logic for adding service-specific information to spans, such as attributes. Specifically, we are adding logic to add
+    the `aws.sns.topic.arn` attribute, to be used to generate AWS_REMOTE_RESOURCE_TYPE and AWS_REMOTE_RESOURCE_IDENTIFIER.
+    There exists SpanAttributes.MESSAGING_DESTINATION_NAME in the upstream logic that we could re-purpose here. However,
+    we are not using it here to maintain consistent naming patterns with other AWS resources.
```

Comment:
Synced with @mxiamxia about this. Mentioned we should not use the upstream logic to maintain consistent naming patterns for other AWS resources